packages/mongo-nav/src/mongo-select/MongoSelect.tsx

```diff
@@ -115,28 +115,61 @@ interface OrganizationMongoSelectProps extends BaseMongoSelectProps {
   isOnPrem?: boolean;
 }
 
-const onKeyDown: React.KeyboardEventHandler = e => {
-  // Stops default browser behavior from automatically scrolling the component
-  if ([keyMap.ArrowUp, keyMap.ArrowDown].includes(e.keyCode)) {
-    e.preventDefault();
-  }
-};
-
 function OrgSelect({
   current,
   data,
   urls,
   isActive,
-  onChange,
+  onChange: onChangeProp,
   onClick,
   constructOrganizationURL,
   isOnPrem,
   disabled,
   loading = false,
 }: OrganizationMongoSelectProps) {
+  const [renderedData, setRenderedData] = useState(data);
+  const [value, setValue] = useState('');
   const [open, setOpen] = useState(false);
   const onElementClick = useOnElementClick();
 
+  useEffect(() => {
+    setRenderedData(data);
+  }, [data]);
+
+  const onChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setValue(e.target.value);
+
+    if (onChangeProp) {
+      onChangeProp(e);
```

Comment:
So basically we only filter internally if the user doesn't set `onChangeProp`, right?

I think two things are possible to improve here:

- Abstract the filter logic into a standalone function that can be shared
- Not have to create an intermediate state value of `renderedData`. I think you can do this all inline in the component body.

Happy to pair on this.

Reply:
👍